[BUG] Panic while suspending namespace

Executing command ` kubectl suspender suspend echo-server` results with:
```
9:59AM INF  version '0.1.0' (built: 2022-05-11T13:04:15Z)
9:59AM INF list of namespaces that will be suspended: [echo-server]
⠋ suspending namespace echo-serverE0512 09:59:54.165540   21459 runtime.go:79] Observed a panic: "assignment to entry in nil map" (assignment to entry in nil map)
goroutine 1 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x137fb40?, 0x1710360})
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.24.0/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x7f649e4c0250?})
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.24.0/pkg/util/runtime/runtime.go:49 +0x75
panic({0x137fb40, 0x1710360})
	/opt/hostedtoolcache/go/1.18.1/x64/src/runtime/panic.go:838 +0x207
github.com/govirtuo/kubectl-suspender/app.App.UpdateNamespace.func1()
	/home/runner/work/kubectl-suspender/kubectl-suspender/app/app.go:94 +0x1df
k8s.io/client-go/util/retry.OnError.func1()
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.24.0/util/retry/util.go:51 +0x33
k8s.io/apimachinery/pkg/util/wait.ConditionFunc.WithContext.func1({0x13b81e0, 0x1})
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.24.0/pkg/util/wait/wait.go:220 +0x1b
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtectionWithContext({0x1725518?, 0xc0000460c8?}, 0x30?)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.24.0/pkg/util/wait/wait.go:233 +0x57
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection(0x1403a40?)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.24.0/pkg/util/wait/wait.go:226 +0x39
k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff({0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0}, 0x40d687?)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.24.0/pkg/util/wait/wait.go:421 +0x5f
k8s.io/client-go/util/retry.OnError({0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0}, 0x15e5f60, 0xc0003d6f00)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.24.0/util/retry/util.go:50 +0xf1
k8s.io/client-go/util/retry.RetryOnConflict(...)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.24.0/util/retry/util.go:104
github.com/govirtuo/kubectl-suspender/app.App.UpdateNamespace({{{0x1719a18, 0xc000478b60}, 0xff, {0x0, 0x0}, {0xc0000b2600, 0x1, 0x1f4}, {0xc000478b70, 0x1, ...}, ...}, ...}, ...)
	/home/runner/work/kubectl-suspender/kubectl-suspender/app/app.go:84 +0x17a
github.com/govirtuo/kubectl-suspender/cmd.glob..func2(0x20d6600?, {0xc00038f6e0?, 0x1?, 0x1?})
	/home/runner/work/kubectl-suspender/kubectl-suspender/cmd/suspend.go:84 +0xc74
github.com/spf13/cobra.(*Command).execute(0x20d6600, {0xc00038f680, 0x1, 0x1})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:860 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0x20d6380)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:902
github.com/govirtuo/kubectl-suspender/cmd.Execute(0xc00034a540?)
	/home/runner/work/kubectl-suspender/kubectl-suspender/cmd/root.go:25 +0xa7
main.main()
	/home/runner/work/kubectl-suspender/kubectl-suspender/main.go:19 +0x11b
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map

goroutine 1 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x7f649e4c0250?})
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.24.0/pkg/util/runtime/runtime.go:56 +0xd8
panic({0x137fb40, 0x1710360})
	/opt/hostedtoolcache/go/1.18.1/x64/src/runtime/panic.go:838 +0x207
github.com/govirtuo/kubectl-suspender/app.App.UpdateNamespace.func1()
	/home/runner/work/kubectl-suspender/kubectl-suspender/app/app.go:94 +0x1df
k8s.io/client-go/util/retry.OnError.func1()
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.24.0/util/retry/util.go:51 +0x33
k8s.io/apimachinery/pkg/util/wait.ConditionFunc.WithContext.func1({0x13b81e0, 0x1})
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.24.0/pkg/util/wait/wait.go:220 +0x1b
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtectionWithContext({0x1725518?, 0xc0000460c8?}, 0x30?)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.24.0/pkg/util/wait/wait.go:233 +0x57
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection(0x1403a40?)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.24.0/pkg/util/wait/wait.go:226 +0x39
k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff({0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0}, 0x40d687?)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.24.0/pkg/util/wait/wait.go:421 +0x5f
k8s.io/client-go/util/retry.OnError({0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0}, 0x15e5f60, 0xc0003d6f00)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.24.0/util/retry/util.go:50 +0xf1
k8s.io/client-go/util/retry.RetryOnConflict(...)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.24.0/util/retry/util.go:104
github.com/govirtuo/kubectl-suspender/app.App.UpdateNamespace({{{0x1719a18, 0xc000478b60}, 0xff, {0x0, 0x0}, {0xc0000b2600, 0x1, 0x1f4}, {0xc000478b70, 0x1, ...}, ...}, ...}, ...)
	/home/runner/work/kubectl-suspender/kubectl-suspender/app/app.go:84 +0x17a
github.com/govirtuo/kubectl-suspender/cmd.glob..func2(0x20d6600?, {0xc00038f6e0?, 0x1?, 0x1?})
	/home/runner/work/kubectl-suspender/kubectl-suspender/cmd/suspend.go:84 +0xc74
github.com/spf13/cobra.(*Command).execute(0x20d6600, {0xc00038f680, 0x1, 0x1})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:860 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0x20d6380)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:902
github.com/govirtuo/kubectl-suspender/cmd.Execute(0xc00034a540?)
	/home/runner/work/kubectl-suspender/kubectl-suspender/cmd/root.go:25 +0xa7
main.main()
	/home/runner/work/kubectl-suspender/kubectl-suspender/main.go:19 +0x11b
```

Namespace exists:
```
❯ kubectl get ns | grep echo-server
echo-server             Active   25h
```

And deployment is there:
```
❯ kubectl get deploy -n echo-server
NAME          READY   UP-TO-DATE   AVAILABLE   AGE
echo-server   1/1     1            1           25h
```

--- a/cmd/suspend.go
+++ b/cmd/suspend.go
@@ -81,6 +81,19 @@ Usage examples:
 			status := "done\n"
 			s := utils.CreateSpinner("suspending namespace " + n)
 			s.Start()
+
+			ns, err := a.Clientset.CoreV1().Namespaces().Get(context.TODO(), n, metav1.GetOptions{})
+			if err != nil {
+				a.Logger.Error().Err(err).Msgf("cannot get namespace '%s'", n)
+				s.Stop()
+				continue
+			}
+			if err := utils.IsNamespaceWatched(ns, a.AnnotationsNames.Prefix+"/"+a.AnnotationsNames.ControllerName, a.ControllerName); err != nil {
+				a.Logger.Error().Err(err).Msgf("namespace '%s' is not watched", n)
+				s.Stop()
+				continue
+			}
+
 			if err := a.UpdateNamespace(n, app.SuspendedState); err != nil {
 				a.Logger.Error().Err(err).Msgf("cannot suspend namespace '%s'", n)
 				status = "failed\n"

--- a/cmd/unsuspend.go
+++ b/cmd/unsuspend.go
@@ -76,6 +76,19 @@ Usage example:
 			status := "done\n"
 			s := utils.CreateSpinner("unsuspending namespace " + n)
 			s.Start()
+
+			ns, err := a.Clientset.CoreV1().Namespaces().Get(context.TODO(), n, metav1.GetOptions{})
+			if err != nil {
+				a.Logger.Error().Err(err).Msgf("cannot get namespace '%s'", n)
+				s.Stop()
+				continue
+			}
+			if err := utils.IsNamespaceWatched(ns, a.AnnotationsNames.Prefix+"/"+a.AnnotationsNames.ControllerName, a.ControllerName); err != nil {
+				a.Logger.Error().Err(err).Msgf("namespace '%s' is not watched", n)
+				s.Stop()
+				continue
+			}
+
 			if err := a.UpdateNamespace(n, app.RunningState); err != nil {
 				a.Logger.Error().Err(err).Msgf("cannot unsuspend namespace '%s'", n)
 				status = "failed\n"

--- a/utils/utils.go
+++ b/utils/utils.go
@@ -1,6 +1,7 @@
 package utils
 
 import (
+	"fmt"
 	"time"
 
 	"github.com/briandowns/spinner"
@@ -40,4 +41,15 @@ func CreateSpinner(suffix string) *spinner.Spinner {
 	s := spinner.New(spinner.CharSets[14], 100*time.Millisecond)
 	s.Suffix = " " + suffix
 	return s
+}
+
+func IsNamespaceWatched(ns *v1.Namespace, controllerNameFullAnnot, controllerName string) error {
+	val, ok := ns.Annotations[controllerNameFullAnnot]
+	if !ok {
+		return fmt.Errorf("this namespace is not watched by kube-ns-suspender")
+	}
+	if val != controllerName {
+		return fmt.Errorf("namespace's controller name does not match the configured controller name")
+	}
+	return nil
 }

